docs(metrics): Add search bar documentation

DominikB2014 opened the pull request:
For MEP, our search bar functionality will have to change, this PR documents the changes.
It includes a
1. A summary of the changes/functionality under "performance monitoring > filters and display"
![image](https://user-images.githubusercontent.com/44422760/183463358-927d6504-84ac-4885-b813-ad2c2a45adf0.png)

2. A link/alert from 2 locations of the documentation, "performance monitoring" and "sentry basics > search" (because performance search is mentioned here as well).
![image](https://user-images.githubusercontent.com/44422760/183463984-79dfc1c4-219e-498d-b886-bf132f243d38.png)

DominikB2014 committed:
Add search bar documentation

DominikB2014 committed:
Fix typos

DominikB2014 committed:
Updated docs

DominikB2014 committed:
Fix typo

0Calories commented on src/docs/product/performance/filters-display/index.mdx:
```diff
@@ -38,6 +38,23 @@ When you select "All Transactions", the table includes all transactions across a
+
+![Updated transaction search](transaction-name-search.png)
+
+When you begin typing a list of transaction names that match your search appear. There are two ways to interact with the search results.
```
```suggestion
When you begin typing, a list of transaction names that match your search appear. There are two ways to interact with the search results:
```

0Calories commented on src/docs/product/performance/filters-display/index.mdx:
```diff
@@ -38,6 +38,23 @@ When you select "All Transactions", the table includes all transactions across a
+
+When you begin typing a list of transaction names that match your search appear. There are two ways to interact with the search results.
+
+1. Clicking on a transaction name will apply a filter on the performance page to only display transaction data for the selected transaction.
```
```suggestion
1. Clicking on a transaction name will apply a filter on the Performance page to only display transaction data for the selected transaction.
```

0Calories commented on src/docs/product/performance/filters-display/index.mdx:
```diff
@@ -38,6 +38,23 @@ When you select "All Transactions", the table includes all transactions across a
+When you begin typing a list of transaction names that match your search appear. There are two ways to interact with the search results.
+
+1. Clicking on a transaction name will apply a filter on the performance page to only display transaction data for the selected transaction.
+2. Clicking the upwards arrow on a search result will take you to that transactions summary page.
```
```suggestion
2. Clicking the upwards arrow icon on a search result will take you to that transaction's summary page.
```

0Calories commented on src/docs/product/performance/index.mdx:
```diff
@@ -26,6 +26,12 @@ Using the information on this page, you can trace issues back through services (
 
+<Alert level="info" title="Important">
+
+For organizations participating in the limited release, the transaction filter on the performance page has been simplified to support metric enhanced performance. The search bar can only be used to filter by transaction name. See [transaction search](/product/performance/filters-display/#transaction-search).
```
```suggestion
For organizations participating in the limited release, the transaction filter on the Performance page has been simplified to support metric enhanced performance. The search bar can only be used to filter by transaction name. See [transaction search](/product/performance/filters-display/#transaction-search).
```

Also, I think Metrics Enhanced Performance is a term that we use internally only, so maybe we should say something else here instead?

DominikB2014 committed:
Apply suggestions from code review

Co-authored-by: Ash Anand <0Calories@users.noreply.github.com>

nanoburd commented:
Can you replace "metrics enhanced performance" with "metrics" and link to https://docs.sentry.io/product/sentry-basics/metrics/

nanoburd commented:
Please also include the beta banner. That you see in other documentation that's already in prod.

gggritso commented on src/docs/product/sentry-basics/search/index.mdx:
```diff
@@ -12,6 +12,12 @@ Search is available on several features throughout [sentry.io](https://sentry.io
 
 ## Query Syntax {#syntax}
 
+<Alert level="info" title="Important">
+
+For organizations participating in the limited release, the transaction filter on the performance page has been simplified to support metric enhanced performance. The search bar can only be used to filter by transaction name. See [transaction search](/product/performance/filters-display/#transaction-search).
+
+</Alert>
+
```
I think it would be good to deemphasize this by moving it lower in the document. The transaction search limitation is only in one place, and it only affects the raw string! No need to scare people at the very top of the page. Maybe this could live under the "The token `example error`" paragraph?

DominikB2014 replied:
I was mixed between putting it at the top, the reason I put it at the top is because it is what you see when you click "read the docs" in the search bar.
![image](https://user-images.githubusercontent.com/44422760/183485858-c8fdce20-f1d7-442c-af3e-b3daf70814e0.png)

So it might be confusing for a user, if they click read the docs while searching for a transaction, only to figure out later in the documentation that the query syntax doesn't apply.

gggritso commented on src/docs/product/performance/index.mdx:
```diff
@@ -26,6 +26,12 @@ Using the information on this page, you can trace issues back through services (
 
 ## Filter Performance Data
 
+<Alert level="info" title="Important">
+
+For organizations participating in the limited release, the transaction filter on the Performance page has been simplified to support metric enhanced performance. The search bar can only be used to filter by transaction name. See [transaction search](/product/performance/filters-display/#transaction-search).
+
+</Alert>
+
```
Might be good to clarify the limited release, and to explain the difference from other searches! Maybe something like

"The transaction filter has been simplified for organizations participating in the limited release of Metrics Enhanced Performance. See blah blah"

Then the "transaction search" section can go into more detail

DominikB2014 replied:
Yeah I do agree, I was aiming to be consistent with the following documentation, as we have a standard blurb before any beta feature
https://sentry.slack.com/archives/C02N6582ZFA/p1659603015550699?thread_ts=1659573427.357199&cid=C02N6582ZFA

I feel like we'd have to update the blurb everywhere.

gggritso replied:
I'm confused, the banner there is different from yours, what's being kept consistent? Also, the banner there is under the "Server-side sampling" heading, which is indeed in limited release. Here, the alert implies that "Filter Performance Data" is in limited release which is ... kind of the case? Maybe PM should tiebreak this one

DominikB2014 replied:
I see the confusion, so this alert I meant to put an alert I wanted to state that this doesn't really apply if your in the MEP release. And to click the link to see the updated functionality for transaction name search.

gggritso commented on src/docs/product/performance/filters-display/index.mdx:
```diff
@@ -38,6 +38,23 @@ When you select "All Transactions", the table includes all transactions across a
 
 "Mobile" puts an emphasis on [Mobile Vitals](/product/performance/mobile-vitals/). Note that if you've selected a non-mobile project, this mode also displays all of those transactions.
 
+## Transaction Search
```
This section feels like it fits nicely under the heading just below! You could rename "Search Condition Filters" to "Transaction Search", and combine them. Both sections are about the search bar, and this `<Note>` could be a caveat

gggritso commented on src/docs/product/performance/filters-display/index.mdx:
```diff
@@ -38,6 +38,23 @@ When you select "All Transactions", the table includes all transactions across a
+
+</Note>
+
+The transaction filter on the performance page has been simplified to support metric enhanced performance. The search bar can only be used to filter by transaction name.
```
Maybe helpful to clarify the difference from the regular search bar. Maybe something like "The search bar can only be used to filter by transaction name, rather than to search for transactions matching the query".

gggritso replied:
Also, I wonder if MEP is a marketing-visible feature, in which case it should be capitalized?

DominikB2014 replied:
As suggested by @nanoburd we replaced the MEP with metrics, as metrics has documentation to link to. Also, I agree, and reworded the search bar docs as suggested!

gggritso reviewed (commented):
Fair's fair! Nits and questions only!

DominikB2014 committed:
Code review

DominikB2014 committed:
Merge branch 'PERF-1619' of github.com:getsentry/sentry-docs into PERF-1619

DominikB2014 committed:
Updated alert blurb

DominikB2014 committed:
Update docs

DominikB2014 committed:
Update index.mdx

DominikB2014 committed:
Update

DominikB2014 commented:
@gggritso and @0Calories Updated docs

As you will either be able to search for transaction name only, or add filters. I've updated the docs so that both the transaction name only search and search condition filters are under the same heading. And the note mentions that if you are not in beta, you will use the condition filter instead.

<img width="709" alt="image" src="https://user-images.githubusercontent.com/44422760/183743115-44f6e9be-fc35-4769-8b1b-0aab594a7f36.png">

Also, changed the alert to a note on the pages that reference search

<img width="711" alt="image" src="https://user-images.githubusercontent.com/44422760/183744489-e08e0edb-06aa-4e65-87ee-354f11999618.png">

gggritso commented:
@DominikB2014 right on, I think this is a lot clearer!

DominikB2014 commented:
@nanoburd I've updated the banner with the changes, let me know what you think!

DominikB2014 committed:
Update index.mdx

DominikB2014 committed:
Update index.mdx

DominikB2014 committed:
Update docs

gggritso reviewed (approved):
🖊️ 👍🏻
